### PR TITLE
References GitHub issue #122

### PR DESCRIPTION
**Pull request by soberman**
Adds support for 'prepare-commit-msg' hook

### CODE REVIEW

**Comment by soberman** [comment]
I AGREE TO THE GITUP CONTRIBUTOR LICENSE AGREEMENT

**Comment by soberman** [comment]
I have just force-pushed a new commit that moves the `prepare-commit-msg` logic into `GICommitViewController` and persists messages up until commit is done.

File: GitUpKit/Core/GCRepository.h [hunk 1]
```diff
@@ -78,5 +78,6 @@ typedef NS_ENUM(NSUInteger, GCFileMode) {
 #if !TARGET_OS_IPHONE
 - (NSString*)pathForHookWithName:(NSString*)name;  // Returns nil if hook doesn't exist
 - (BOOL)runHookWithName:(NSString*)name arguments:(NSArray*)arguments standardInput:(NSString*)standardInput error:(NSError**)error;  // Silently ignores non-existing hooks
+- (BOOL)runHookWithName:(NSString*)name arguments:(NSArray*)arguments standardInput:(NSString*)standardInput standardOutput:(NSString**)standardOutput error:(NSError**)error;
```
Comment (swisspol) [inline GitUpKit/Core/GCRepository.h]: No need to add a new method. Just add the argument to the existing one and update the call sites.

This should be a separate commit BTW, the first one of the PR.

**Comment by swisspol** [comment]
Looking at the rest of this PR, there are other fundamental issues. Unfortunately, I just don't have the time to go over them right now, so let's put this on hold for now.
